A couple simple borrowck cleanups

lqd opened the pull request:
This PR has a couple simple renamings:
- it's been a long time since the mapping from `Location`s to `PointIndex`es was extracted from `RegionElements` into the `DenseLocationMap`, but only the types were renamed at the time. borrowck still refers to this map as `elements`. That's confusing, especially since sometimes we also use the mapping via `LivenessValues`, and makes more sense as `location_map` instead.
- to clarify `LocationTable` is not as general as it sounds, and is only for datalog polonius. In this branch I didn't rename the handful of `location_table` fields and params to `polonius_table`, but can do that to differentiate it even more from `location_map`. I did try it locally and it looks worthwhile, so if you'd prefer I can also push it here. (Or we could even switch these datalog types and fields to even more explicit names)
- to clarify the incomprehensible `AllFacts`, it is renamed to `PoloniusFacts`. These can be referred to as `facts` within the legacy polonius module, but as `polonius_facts` outside of it to make it clear that they're not about NLLs (nor are they about in-tree polonius but that'll be magically fixed when they're removed in the future)

r? @matthewjasper

lqd committed:
stop calling `DenseLocationMap` "elements"

"Elements" are `RegionElement`s. The dense location mapping was removed
from the element containers a while ago but didn't rename its use-sites.
Most of the old naming only used the mapping, and are better named
`location_map`.

lqd committed:
rename `LocationTable` to `PoloniusLocationTable`

Its original naming hides the fact that it's related to datalog
polonius, and bound to be deleted in the near future.

It also conflicts with the expected name for the actual NLL location
map, and prefixing it with its use will make the differentiation
possible.

rust-log-analyzer commented:
The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12670246968/job/35309514136) [(plain)](https://github.com/rust-lang/rust/commit/921e4073e013fc8d558c8541ba3f2e16f3e8302a/checks/35309514136/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
 Documenting rustc-main v0.0.0 (/checkout/compiler/rustc)
error: unresolved link to `LocationTable`
  --> compiler/rustc_borrowck/src/consumers.rs:36:41
   |
36 |     /// and additionally retrieve the [`LocationTable`] and [`PoloniusInput`] that
   |
   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
   = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`
   = help: to override `-D warnings` add `#[allow(rustdoc::broken_intra_doc_links)]`

error: could not document `rustc_borrowck`
warning: build failed, waiting for other jobs to finish...
Command has failed. Rerun with -v to see more details.
  local time: Wed Jan  8 12:48:09 UTC 2025
  network time: Wed, 08 Jan 2025 12:48:09 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

lqd committed:
rename `AllFacts` to `PoloniusFacts`

This is another strangely named struct (and associated fields) that is
hard to see was related to datalog polonius.

matthewjasper commented:
@bors r+

bors commented:
:pushpin: Commit 36ea00c20d24f3eafdaa5b79c2c16f28c2b4a76e has been approved by `matthewjasper`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=matthewjasper 36ea00c20d24f3eafdaa5b79c2c16f28c2b4a76e -->
<!-- homu: {"type":"Approved","sha":"36ea00c20d24f3eafdaa5b79c2c16f28c2b4a76e","approver":"matthewjasper","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
